### PR TITLE
Removed static from urlCacheControlMap and made it thread safe

### PR DESCRIPTION
**Pull request by mgherghe**
This was only tested by reloading the demo page several times and checking that the urlCacheControlMap is properly populated.
All tests from the gateway.service.http.directory module passed.

### CODE REVIEW

**Comment by dpwspoon** [comment]
Do we want this on the release branch?

File: service/http.directory/src/main/java/org/kaazing/gateway/service/http/directory/HttpDirectoryServiceHandler.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
     private boolean indexes;
 
     private List<PatternCacheControl> patterns;
-    private static Map<String, CacheControlHandler> urlCacheControlMap = new HashMap<String, CacheControlHandler>();
+    private Map<String, CacheControlHandler> urlCacheControlMap = new ConcurrentHashMap<String, CacheControlHandler>();
```
Comment (jitsni) [inline service/http.directory/src/main/java/org/kaazing/gateway/service/http/directory/HttpDirectoryServiceHandler.java]: OK, static is gone. But just using ConcurrentHashMap itself doesn't give thread-safety for composite operations (for e.g contains and put). You may want to try something like the following. The idea is to compute CacheControlHandler atomically (when it is already in the map or not). See if it works :

```
    private void addCacheControl(HttpAcceptSession session, File requestFile, String requestPath) {
        CacheControlHandler cacheControlHandler = urlCacheControlMap.computeIfAbsent(requestPath, path ->
             patterns.stream()
                     .filter(patternCacheControl -> PatternMatcherUtils.caseInsensitiveMatch(path, patternCacheControl.getPattern()))
                     .findFirst()
                     .map(patternCacheControl -> new CacheControlHandler(requestFile, patternCacheControl))
                     .orElse(null)
        );

        if (cacheControlHandler != null) {
            addCacheControlHeader(session, requestFile, cacheControlHandler);
        }
}
```

I don't see any operations removing from the map (except clearing the whole thing). Don't the entries in the map be removed once they are expired ?

**Comment by jitsni** [comment]
Also, do you want this in the release ?

File: service/http.directory/src/main/java/org/kaazing/gateway/service/http/directory/HttpDirectoryServiceHandler.java (hunk 1 above)
Comment (mgherghe) [inline service/http.directory/src/main/java/org/kaazing/gateway/service/http/directory/HttpDirectoryServiceHandler.java]: The map is used to match the URL of a requested file with a pattern from the configuration file. That is only done once and needs to be cached, so there is no need to remove the entries. 
The max-age and any other information that may change is computed at the request time.

**Comment by mgherghe** [comment]
Recreated [PR](https://github.com/kaazing/gateway/pull/526) against `release/5.1.0` branch.

**Pull request closed without merging**